### PR TITLE
Facebook login

### PR DESCRIPTION
**Pull request by jacksonargo**

### CODE REVIEW

File: pkg/facebook/facebook.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	return "https://www.facebook.com/v8.0/dialog/oauth?" + query.Encode()
+}
+
+type OAuthToken struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+// Query oauth token from discord.
```
Comment (nightlark) [inline pkg/facebook/facebook.go:66]:
```suggestion
// Query oauth token from facebook.
```

File: pkg/api/testdata/login.html [hunk 2]
```diff
@@ -56,6 +56,7 @@
         <button class="btn btn-lg btn-block bg-vvgo-purple text-light" type="submit">Sign in</button>
     </form>
     <a class="mt-1 btn btn-lg btn-block bg-discord-blue text-light" href="/login/discord">Sign in with Discord</a>
+    <a class="mt-1 btn btn-lg btn-block bg-discord-blue text-light" href="/login/facebook">Sign in with Facebook</a>
```
Comment (nightlark) [inline pkg/api/testdata/login.html:59]:
```suggestion
    <a class="mt-1 btn btn-lg btn-block bg-facebook-blue text-light" href="/login/facebook">Login with Facebook</a>
```

File: public/login.gohtml [hunk 3]
```diff
@@ -34,6 +34,7 @@
         <button class="btn btn-lg btn-block bg-vvgo-purple text-light" type="submit">Sign in</button>
     </form>
     <a class="mt-1 btn btn-lg btn-block bg-discord-blue text-light" href="/login/discord">Sign in with Discord</a>
+    <a class="mt-1 btn btn-lg btn-block bg-discord-blue text-light" href="/login/facebook">Sign in with Facebook</a>
```
Comment (nightlark) [inline public/login.gohtml:37]:
```suggestion
    <a class="mt-1 btn btn-lg btn-block bg-facebook-blue text-light" href="/login/facebook">Login with Facebook</a>
```

**Comment by jacksonargo** [comment]
I talked it over with the leaders, and due to the high amount of gating facebook puts around it's api, I'm actually gonna toss this. Facebook has access to a lot of personal data, and we really don't need access to all that. Majority of our members are now on discord anyways, so it's not such a big deal anyways.

**Comment by jacksonargo** [comment]
I am going to keep the abstractions tho and put them in a different pr

**Pull request closed without merging**